src/librustc_borrowck/diagnostics.rs

```diff
@@ -160,6 +160,35 @@ fn main(){
 ```
 "##,
 
+E0386: r##"
+This error occurs when an attempt is made to mutate the target of a mutable
+reference stored inside an immutable container.
+
+For example, this can happen when storing a `&mut` inside an immutable `Box`:
+
+```
+let mut x: i64 = 1;
+let y: Box<_> = Box::new(&mut x);
+**y = 2; // error, cannot assign to data in an immutable container
+```
+
+This error can be fixed by making the container mutable:
+
+```
+let mut x: i64 = 1;
+let mut y: Box<_> = Box::new(&mut x);
+**y = 2;
+```
+
+It can also be fixed by using a type with interior mutability, such as `Cell` or
+`RefCell`:
+
+```
+let y: Cell<_> = Cell::new(1);
```

Comment:
Go along the lines of the example above, `let y: Box<Cell<_>> = Box::new(Cell::new(x))`, and then `y.set(2)`